### PR TITLE
Add Checks to Ensure Correspondence Auto Assignment Job Uniqueness

### PR DESCRIPTION
**Pull request by tradesmanhelix**
Resolves [Auto Assign Correspondence Job Check](https://jira.devops.va.gov/browse/APPEALS-40934)

# Description

- Update the CorrespondenceAutoAssigner to do validations before running to ensure we don't end up with parallel auto assign jobs.

- Add and update specs as needed.

## Acceptance Criteria
- [ ] All specs passing

## Testing Plan
1. Verify specs cover main use case and edge cases.

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### CODE REVIEW

File: app/controllers/correspondence_controller.rb [hunk 1]
```diff
@@ -175,26 +175,19 @@ def process_intake
...
-      }
-
-      begin
-        perform_later_or_now(AutoAssignCorrespondenceJob, job_args)
-      ensure
-        render json: { batch_auto_assignment_attempt_id: batch.id }, status: :ok
-      end
-    else
-      render json: { batch_auto_assignment_attempt_id: batch.id }, status: :ok
-    end
```
Comment (tradesmanhelix) [inline app/controllers/correspondence_controller.rb:197]: This code had a bug where if a `BatchAutoAssignment` got stuck in `started`, you couldn't run another one.

This code is now moved into the service's logic and is not needed here.

File: app/jobs/auto_assign_correspondence_job.rb [hunk 2]
```diff
@@ -4,12 +4,15 @@ class AutoAssignCorrespondenceJob < CaseflowJob
   queue_with_priority :low_priority
 
   def perform(current_user_id:, batch_auto_assignment_attempt_id:)
-    CorrespondenceAutoAssigner.new(
+    correspondence_auto_assigner.perform(
       current_user_id: current_user_id,
       batch_auto_assignment_attempt_id: batch_auto_assignment_attempt_id
-    ).perform
-  rescue StandardError => error
```
Comment (tradesmanhelix) [inline app/jobs/auto_assign_correspondence_job.rb:11]: Don't rescue errors. We want to allow them to bubble up so that the queue will retry the job if it fails. We'll handle graceful exits from unexpected errors in the class itself.

File: app/services/correspondence_auto_assign_logger.rb
Comment (tradesmanhelix) [inline app/services/correspondence_auto_assign_logger.rb:1]: Updated to use `attr_accessor`.

File: app/services/correspondence_auto_assigner.rb [hunk 3]
```diff
@@ -56,38 +54,43 @@ def assign_task_to_user(task, user)
...
       assigned_at: Time.current,
-      assigned_by: @current_user,
+      assigned_by: current_user,
       assigned_to_type: User.name,
       status: Constants.TASK_STATUSES.assigned
     )
   end
 
   def unassigned_review_package_tasks
-    return @unassigned_review_package_tasks if @unassigned_review_package_tasks.present?
```
Comment (tradesmanhelix) [inline app/services/correspondence_auto_assigner.rb:66]: Need to use DB as source of truth, so can't use caching here.

**Review by cacevesva — APPROVED** [review]
